### PR TITLE
Added Xiaomi Miija door/window bluetooth sensor

### PR DESCRIPTION
**Pull request by sanchosk**
Sensor displays Button value.
Needs atc unlocking first.

## Description:

**Related issue (if applicable):** fixes #<Tasmota issue number goes here>

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [ ] The code change is tested and works on Tasmota core ESP8266 V.2.7.4.9
  - [x] The code change is tested and works on Tasmota core ESP32 V.1.0.5-rc6
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/xsns_62_esp32_mi_ble.ino [hunk 1]
```diff
@@ -369,8 +369,9 @@ void (*const MI32_Commands[])(void) PROGMEM = {
 #define MI_MHOC401     11
 #define MI_MHOC303     12
 #define MI_ATC         13
+#define MI_DOOR        14
 
-#define MI_MI32_TYPES    13 //count this manually
+#define MI_MI32_TYPES    14 //count this manually
```
Comment (sanchosk) [inline tasmota/xsns_62_esp32_mi_ble.ino:374]: Adding the new sensor type to the list

File: tasmota/xsns_62_esp32_mi_ble.ino [hunk 2]
```diff
@@ -385,7 +386,8 @@ const uint16_t kMI32DeviceID[MI_MI32_TYPES]={
   0x0153, // yee-rc
   0x0387, // MHO-C401
   0x06d3, // MHO-C303
-  0x0a1c  // ATC -> this is a fake ID
+  0x0a1c,  // ATC -> this is a fake ID
+  0x098b  // door/window sensor
```
Comment (sanchosk) [inline tasmota/xsns_62_esp32_mi_ble.ino:390]: Adding new type to the array

File: tasmota/xsns_62_esp32_mi_ble.ino [hunk 3]
```diff
@@ -401,7 +403,8 @@ const char kMI32DeviceType9[] PROGMEM = "YEERC";
 const char kMI32DeviceType10[] PROGMEM ="MHOC401";
 const char kMI32DeviceType11[] PROGMEM ="MHOC303";
 const char kMI32DeviceType12[] PROGMEM ="ATC";
-const char * kMI32DeviceType[] PROGMEM = {kMI32DeviceType0,kMI32DeviceType1,kMI32DeviceType2,kMI32DeviceType3,kMI32DeviceType4,kMI32DeviceType5,kMI32DeviceType6,kMI32DeviceType7,kMI32DeviceType8,kMI32DeviceType9,kMI32DeviceType10,kMI32DeviceType11,kMI32DeviceType12};
+const char kMI32DeviceType13[] PROGMEM ="DOOR";
+const char * kMI32DeviceType[] PROGMEM = {kMI32DeviceType0,kMI32DeviceType1,kMI32DeviceType2,kMI32DeviceType3,kMI32DeviceType4,kMI32DeviceType5,kMI32DeviceType6,kMI32DeviceType7,kMI32DeviceType8,kMI32DeviceType9,kMI32DeviceType10,kMI32DeviceType11,kMI32DeviceType12,kMI32DeviceType13};
```
Comment (sanchosk) [inline tasmota/xsns_62_esp32_mi_ble.ino:407]: Wasn't able to find short and relevant name, so I used "DOOR" :(

File: tasmota/xsns_62_esp32_mi_ble.ino [hunk 4]
```diff
@@ -1416,6 +1419,7 @@ uint32_t MIBLEgetSensorSlot(const uint8_t *mac, uint16_t _type, uint8_t counter)
       _newSensor.feature.bat=1;
       break;
     case MI_YEERC:
+    case MI_DOOR:
       _newSensor.feature.Btn=1;
```
Comment (sanchosk) [inline tasmota/xsns_62_esp32_mi_ble.ino:1423]: Adding a "button" feature to track the magnetic sensor value.

File: tasmota/xsns_62_esp32_mi_ble.ino [hunk 5]
```diff
@@ -1761,6 +1765,12 @@ int MI32parseMiPayload(int _slot, struct mi_beacon_data_t *parsed){
       // AddLog(LOG_LEVEL_DEBUG,PSTR("M32: Mode 17: NMT: %u seconds"), _beacon.NMT);
     } break;
 
+    case 0x19:{
+      MIBLEsensors[_slot].Btn = uint8_t(parsed->payload.data[0]); // just an 8 bit value in a union.
+      MIBLEsensors[_slot].eventType.Btn = 1;
+      MI32.mode.shallTriggerTele = 1;
+    } break;
+
```
Comment (sanchosk) [inline tasmota/xsns_62_esp32_mi_ble.ino:1773]: Actual code for device type 0x19. Not sure how else can I read the payload...

File: tasmota/xsns_62_esp32_mi_ble.ino [hunk 6]
```diff
@@ -2891,7 +2901,7 @@ void MI32Show(bool json)
       if(p->bat!=0x00){
           WSContentSend_PD(HTTP_BATTERY, typeName, p->bat);
       }
-      if (p->type==MI_YEERC){
+      if (p->type==MI_YEERC || p->type==MI_DOOR){
         WSContentSend_PD(HTTP_LASTBUTTON, typeName, p->Btn);
```
Comment (sanchosk) [inline tasmota/xsns_62_esp32_mi_ble.ino:2905]: Making sure the button feature will be reported to UI.

**Comment by btsimonh** [comment]
@arendst - I want to build on this;  are we good to go here?

**Pull request merged**